### DISCUSSION TITLE
Can't understand the syntax in some Trame examples

### DISCUSSION THREAD

**Discussion by kerim371** [Kitware/trame#588] [General]
Hi,

There is a very attractive [Trame example](https://examples.vtk.org/site/Trame/Applications/RemoteSelection/RemoteSelection/) for integrating VTK with Plotly.

There I can see lines like:
```python
                    view = vtk.VtkRemoteView(
                        render_window,
                        box_selection=("vtk_selection",),
                        box_selection_change=(on_box_selection_change, "[$event]"),
                        **VTK_VIEW_SETTINGS,
                    )
```

And:
```python
                        html_plot = plotly.Figure(
                            selected=(
                                on_chart_selection,
                                "[$event?.points.map(({pointIndex}) => pointIndex)]",
                            ),
                            **CHART_STYLE,
                        )
```

What `$` means? or `$event?` ? Does `event` is accessible from python to view its methods (like `dir(event)`) and its type (like `type(event)`)?
How to learn to make such structures of code by myself?

**Comment by jourdain** [comment]
$event comes from [here](https://vuejs.org/guide/essentials/event-handling#accessing-event-argument-in-inline-handlers)

In this context, `selected=` is a triggered event on the client side which is captured in the [summary here](https://kitware.github.io/trame/guide/intro/cheatsheet.html#summary)

As you don't necessary know what the `$event` contains, you can start by doing `something_that_trigger_an_event="console.log($event)"` and then use the dev console of your browser to explore that `$event` object so you can extract only the piece you need for the server side using inline JS expression.

**Comment by kerim371** [comment]
@jourdain thank you!
Good to know.

Haven't work with java-script before and it is very interesting.

P.S. I'm sorry for the late reply
